impeller/core/shader_types.h

```diff
@@ -79,10 +79,20 @@ struct ShaderMetadata {
   std::vector<ShaderStructMemberMetadata> members;
 };
 
+/// @brief Metadata required to bind a buffer.
+///
+/// OpenGL binding requires the usage of the separate shader metadata struct.
 struct ShaderUniformSlot {
+  /// @brief The name of the uniform slot.
   const char* name;
+
+  /// @brief `ext_res_0` is the Metal binding value.
   size_t ext_res_0;
+
+  /// @brief The Vulkan descriptor set index.
   size_t set;
+
+  /// @brief The Vulkan binding value.
   size_t binding;
```

Comment:
Going by the convention in the sampled image slot, ext_res_0 should be in binding for MSL instead of as a separate field 😞